Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp

```diff
@@ -170,24 +170,33 @@ class pfPatcherStream : public plZlibStream
         fOutput = new hsRAMStream;
     }
 
-    pfPatcherStream(pfPatcherWorker* parent, const plFileName& filename, const NetCliFileManifestEntry& entry)
-        : fParent(parent), fFlags(entry.flags), fBytesWritten(0)
+    pfPatcherStream(pfPatcherWorker* parent, const plFileName& reqName, const plFileName& cliName, const NetCliFileManifestEntry& entry)
+        : fParent(parent), fFilename(cliName.Normalize()), fFlags(entry.flags), fBytesWritten(0)
     {
         // ugh. eap removed the compressed flag in his fail manifests
-        if (filename.GetFileExt().compare_i("gz") == 0) {
+        if (reqName.GetFileExt().compare_i("gz") == 0) {
             fFlags |= pfPatcherWorker::kFlagZipped;
             parent->fTotalBytes += entry.zipSize;
         } else
             parent->fTotalBytes += entry.fileSize;
     }
 
-    virtual bool Open(const plFileName& filename, const char* mode)
+    void Begin()
     {
-        fFilename = filename.Normalize();
-        return plZlibStream::Open(fFilename, mode);
+        fDLStartTime = hsTimer::GetSysSeconds();
+        if (!fOutput)
+            Open(fFilename, "wb");
     }
 
-    virtual uint32_t Write(uint32_t count, const void* buf)
+    bool Open(const plFileName& filename, const char* mode) HS_OVERRIDE
```

Comment:
I'll update it to use the parameter instead however. Could be useful in the future